### PR TITLE
Add Encryption Passphrase Check in Installers

### PR DESCRIPTION
**Pull request by niontive**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Add encryption passphrase check in terminal and GUI installers.

Cherry-picked from [PR 91](https://github.com/microsoft/CBL-Mariner/pull/91)

###### Change Log  

- Add "dos2unix" and "words" packages.
- Add "words" package to ISO terminal installer.
- Add password quality logic to "Encrypt System" widget for GUI installer.

###### Does this affect the toolchain?  

No

###### Associated issues  

None

###### Links to CVEs  
None

###### Test Methodology

Local build

### CODE REVIEW

File: SPECS/dos2unix/dos2unix.spec [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+%doc NEWS.txt README.txt TODO.txt
+%{_bindir}/dos2unix
+%{_bindir}/mac2unix
+%{_bindir}/unix2dos
+%{_bindir}/unix2mac
+%{_mandir}/man1/*.1*
+
+%changelog
+* Tue Aug 25 2020 Nicolas Ontiveros <niontive@microsoft.com> - 7.4.1-4
+- Initial CBL-Mariner import from Fedora 33 (license: MIT)
```
Comment (MateuszMalisz) [inline SPECS/dos2unix/dos2unix.spec:46]: The `License` field mentions BSD though?

File: SPECS/words/words.spec [hunk 2]
```diff
@@ -0,0 +1,229 @@
+Summary:       A dictionary of English words for the /usr/share/dict directory
+Name:          words
+Version:       3.0
+Release:       37%{?dist}
+License:       Public Domain
+# Note that Moby Project officially does not exist any more. The most complete
+# information about the project is in Wikipedia.
+URL:           https://en.wikipedia.org/wiki/Moby_Project
+# Source: http://web.archive.org/web/20060527013227/http://www.dcs.shef.ac.uk/research/ilash/Moby/mwords.tar.Z
```
Comment (MateuszMalisz) [inline SPECS/words/words.spec:9]: nit: move the link directly above the Source0

File: SPECS/words/words.spec [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+#457309 - contains both 'unnecessary' and 'unneccesary'
+#1626689 - linux.words contains "half-embracinghalf-embracingly"
+#1652919 - malformed entry in words file
+Patch0:        words-3.0-typos.patch
+#470921 -"Barack" and "Obama" are not in /usr/share/dict/words
+Patch1:        words-3.0-presidents.patch
+
+%description
+The words file is a dictionary of English words for the
+/usr/share/dict directory. Some programs use this database of
```
Comment (MateuszMalisz) [inline SPECS/words/words.spec:29]: nit: replace absolute path with a macro

File: toolkit/resources/imageconfigs/packagelists/iso-initrd-packages.json [hunk 4]
```diff
@@ -61,6 +61,7 @@
         "usbutils",
         "util-linux",
         "vim",
+        "words",
```
Comment (MateuszMalisz) [inline toolkit/resources/imageconfigs/packagelists/iso-initrd-packages.json:64]: Shouldn't `calamares` depend on `words` instead?

File: SPECS/dos2unix/dos2unix.spec (hunk 1 above)
Comment (MateuszMalisz) [reply SPECS/dos2unix/dos2unix.spec:46]: I had the impression that this mentioned the initial license, not the license of the spec file. Makes sense though :)

File: toolkit/resources/imageconfigs/packagelists/iso-initrd-packages.json (hunk 4 above)
Comment (niontive) [reply toolkit/resources/imageconfigs/packagelists/iso-initrd-packages.json:64]: No. The terminal installer needs words.

**Review by schmittjoseph — APPROVED** [review]

**Pull request merged**